transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/AbstractHttpJob.java

```diff
@@ -104,13 +105,25 @@ protected boolean executeHttp()
     {
         int tries = 1;
         boolean doProxy = false;
+        String site = location.getHost();
         try
         {
             while ( tries > 0 )
             {
                 tries--;
                 try
                 {
+                    // if the site is in memory, go with proxy directly, avoid to query db frequently through isProxySite
+                    if ( getProxySitesCache().contains( site ) )
+                    {
+                        doProxy = true;
+                    }
+                    // if the site is not in memory, query db through isProxySite
+                    else if ( isProxySite( site ) )
```

Comment:
For those sites already in the whitelist, this will check it every request, right ? it's better to avoid that if it's the case.

Reply:
@sswguo yeah, that's the point. I'll try to add another mem cache for whitelist, and any first successful request going to the site without proxy will add into it, the following access on this whitelist site will check the mem cache instead to avoid frequent db query.
I'm doing refactor on https://github.com/Commonjava/path-mapped-storage/pull/121, will amend this point together.